experiments/surface_fluxes_perfect_model/generate_truth.jl

```diff
@@ -0,0 +1,131 @@
+# generate_truth: generate true y, noise and x_inputs
+using Pkg
+experiment_id = "surface_fluxes_perfect_model"
+Pkg.activate("experiments/$experiment_id")
+
+import SurfaceFluxes as SF
+import SurfaceFluxes.Parameters as SFPP
+import SurfaceFluxes.UniversalFunctions as UF
+import Thermodynamics as TD
+using YAML
+import SurfaceFluxes.Parameters: SurfaceFluxesParameters
+using CalibrateAtmos
+
+pkg_dir = pkgdir(CalibrateAtmos)
+experiment_path = "$pkg_dir/experiments/$experiment_id"
+data_path = "$experiment_path/data"
+include("$experiment_path/model_interface.jl")
+include("$experiment_path/observation_map.jl")
+
+FT = Float32
+
+"""
+    generate_profiles(FT)
+
+Generate a set of test profiles for the surface fluxes model. Here we want the profiles to be
+statically stable to be sensitive to the input parameters, `a_m` and `a_b`.
+"""
+function generate_profiles(FT)
+    profiles_sfc = []
+    profiles_int = []
+    struct_to_nt(s) =
+        NamedTuple{propertynames(s)}(map(x -> getfield(s, x), propertynames(s)))
+    for i in 1:20
+        push!(
+            profiles_sfc,
+            struct_to_nt(TestAtmosProfile{FT}(T = FT(280 + i / 10), z = FT(0))),
+        )
+        push!(
+            profiles_int,
+            struct_to_nt(
+                TestAtmosProfile{FT}(T = FT(280.2 + i / 10), z = FT(10)),
+            ),
+        )
+    end
+
+    return profiles_sfc, profiles_int
+end
+Base.@kwdef mutable struct TestAtmosProfile{FT}
+    u::FT = FT(2)
+    v::FT = FT(0)
+    ρ::FT = FT(1)
+    q::FT = FT(0.001)
+    T::FT = FT(300)
+    z::FT = FT(0)
+end
+
+
+"""
+    save_profiles(FT; data_path = "data", x_data_file = "data/surface_fluxes_test_data.jld2")
+
+Save the generated profiles to file.
+"""
+function save_profiles(
+    FT;
+    data_path = "data",
+    x_data_file = "data/surface_fluxes_test_data.jld2",
+)
+
+    mkpath(data_path)
+
+    profiles_sfc, profiles_int = generate_profiles(FT)
+
+    data = Dict(
+        "profiles_sfc" => profiles_sfc[1:end],
+        "profiles_int" => profiles_int[1:end],
+    )
+    JLD2.save(joinpath(data_path, x_data_file), data)
+end
+
+"""
+    synthetic_observed_y(x_inputs; data_path = "data")
+
+Generate synthetic observed y from the model truth.
+"""
+function synthetic_observed_y(x_inputs; data_path = "data", apply_noise = false)
+    config = YAML.load_file("$experiment_path/model_config.yml")
+    config["output_dir"] = data_path
+    y = obtain_ustar(FT, x_inputs, config, return_ustar = true)
+    if apply_noise
+        # add noise to model truth to generate synthetic observations with some observation error
+        Γ = 0.003^2 * I * (maximum(y) - minimum(y))
```

Comment:
```suggestion
        Γ = FT(0.003)^2 * I * (maximum(y) - minimum(y))
```

```
julia> Float64(0.003) * I * Float32(2)
UniformScaling{Float64}
0.006*I

julia> Float32(0.003) * I * Float32(2)
UniformScaling{Float32}
0.006*I
```

Does this matter?

Reply:
Well, all the observables are converted to FT64 anyway (required by the EKI struct) but it's cleaner to include this explicitly, as you suggest.